server/src/storage/commits.ts

```diff
@@ -101,13 +101,22 @@ const updateCommit = async (
         value: 1,
       });
       break;
-    case 'completed':
+    case 'completed': {
       listingUpdateRules.push({
         property: 'numCompleted',
         op: 'add',
         value: 1,
       });
+
+      const affectedListing = await getEntity(LISTING_KIND, affectedListingId);
+      if (affectedListing.numCompleted + 1 === affectedListing.numCommits) {
+        listingUpdateRules.push({
+          property: 'listingStatus',
+          value: 'completed',
+        });
```

Comment:
Do we need `op: 'replace'`?
```suggestion
        listingUpdateRules.push({
          property: 'listingStatus',
          op: 'replace',
          value: 'completed',
        });
```

Reply:
@hzxcaryn It's not necessary because the default for `op` is 'replace'. But if it's not clear then I can add the `op` explicitly.